File: 1284-four-divisors/1284-four-divisors.py
class Solution:
    def sumFourDivisors(self, nums: List[int]) -> int:
        res = 0
        
        for num in nums:
            divisors = set()
            
            for i in range(1, int(num**0.5)+1):
                if not num % i:
                    divisors.add(num//i)
                    divisors.add(i)
                if len(divisors) > 4:
                    break
            
            if len(divisors) == 4:
                res += sum(divisors)
        
        return res

File: 1284-four-divisors/1284-four-divisors.cpp
class Solution {
public:
    int sumFourDivisors(vector<int>& nums) {
        long long value=0;
        for(int i=0;i<nums.size();i++)
        {
            vector<long long int> divisors;
            for(int j=1;j<(int)pow(nums[i],0.5)+1;j++)
            {
                if(nums[i]%j==0)
                {
                    if (j != nums[i] / j) {
                        divisors.push_back(nums[i] / j);
                    }
                    divisors.push_back(j);
                }

                if(divisors.size()>4)
                    break;
            }
            if(divisors.size()==4)
                value=value+accumulate(divisors.begin(),divisors.end(), 0);

        }
        return value;
    }
};